packages/inertia/src/inertia.js

```diff
@@ -68,132 +102,145 @@ export default {
     return this.visitId
   },
 
-  visit(url, { method = 'get', data = {}, replace = false, preserveScroll = false, preserveState = false, only = [], headers = {}} = {}) {
-    progress.start()
+  visit(url, {
+    method = 'get',
+    data = {},
+    replace = false,
+    preserveScroll = false,
+    preserveState = false,
+    only = [],
+    headers = {},
+    onCancelToken = () => ({}),
```

Comment:
If we allow the cancel token to be retrieved, wouldn't it also make sense to allow setting it?

Reply:
What's the use-case here?

Reply:
Preloading ;P

Reply:
Ahh yes, understood. Do you want to PR that maybe once this is done...since it's lower urgency?

Reply:
Sure 👍